aspnetcore/security/enforcing-ssl.md

```diff
@@ -46,78 +47,92 @@ The preceding highlighted code:
->
-> * Set `HttpsRedirectionOptions.HttpsPort`.
-> * Set the `ASPNETCORE_HTTPS_PORT` environment variable.
-> * In development, set an HTTPS URL in *launchsettings.json*.
-> * Configure an HTTPS URL endpoint for [Kestrel](xref:fundamentals/servers/kestrel) or [HTTP.sys](xref:fundamentals/servers/httpsys).
->
-> When Kestrel or HTTP.sys is used as a public-facing edge server, Kestrel or HTTP.sys must be configured to listen on both:
->
-> * The secure port where the client is redirected (typically, 443 in production and 5001 in development).
-> * The insecure port (typically, 80 in production and 5000 in development).
->
-> The insecure port must be accessible by the client in order for the app to receive an insecure request and redirect it to the secure port.
->
-> Any firewall between the client and server must also have the ports open for traffic.
->
-> For more information, see [Kestrel endpoint configuration](xref:fundamentals/servers/kestrel#endpoint-configuration) or <xref:fundamentals/servers/httpsys>.
+### Port configuration
 
-The following highlighted code calls [AddHttpsRedirection](/dotnet/api/microsoft.aspnetcore.builder.httpsredirectionservicesextensions.addhttpsredirection) to configure middleware options:
+A port must be available for the middleware to redirect an insecure request to HTTPS. If no port is available:
 
-[!code-csharp[](enforcing-ssl/sample/Startup.cs?name=snippet2&highlight=14-99)]
+* Redirection to HTTPS doesn't occur.
+* The middleware logs the warning "Failed to determine the https port for redirect."
 
-Calling `AddHttpsRedirection` is only necessary to change the values of `HttpsPort` or `RedirectStatusCode`.
+Specify the HTTPS port using any of the following approaches:
 
-The preceding highlighted code:
+* Set [HttpsRedirectionOptions.HttpsPort](#options).
+* Set the `ASPNETCORE_HTTPS_PORT` environment variable or [https_port Web Host configuration setting](xref:fundamentals/host/web-host#https-port):
 
-* Sets [HttpsRedirectionOptions.RedirectStatusCode](/dotnet/api/microsoft.aspnetcore.httpspolicy.httpsredirectionoptions.redirectstatuscode) to [Status307TemporaryRedirect](/dotnet/api/microsoft.aspnetcore.http.statuscodes.status307temporaryredirect), which is the default value. Use the fields of the [StatusCodes](/dotnet/api/microsoft.aspnetcore.http.statuscodes) class for assignments to `RedirectStatusCode`.
-* Sets the HTTPS port to 5001. The default value is 443.
+  **Key**: `https_port`  
+  **Type**: *string*  
+  **Default**: A default value isn't set.  
+  **Set using**: `UseSetting`  
+  **Environment variable**: `<PREFIX_>HTTPS_PORT` (The prefix is `ASPNETCORE_` when using the [Web Host](xref:fundamentals/host/web-host).)
 
-The following mechanisms set the port automatically:
+  When configuring an <xref:Microsoft.AspNetCore.Hosting.IWebHostBuilder> in `Program`:
```

Comment:
`Program` --> *Program.cs*

Reply:
The rule for `Startup`/*Startup.cs* doesn't apply to `Program`? ... Can't `Program` can be in a file of any name?

Reply:
I could be wrong, but I was under the impression that the file must be named *Program.cs*. `Startup` can be any name you want, so long as `Program.Main` points to it in `UseStartup`. It's worth following up on.

Reply:
I'll run a local test and see what happens ... :boom: or 🌻

Reply:
🌻 ... I changed the name of *Program.cs* to *Program_BLAH.cs*. Result ...

<img width="291" alt="capture" src="https://user-images.githubusercontent.com/1622880/47332836-4c3e3480-d646-11e8-8ede-27f24a7cda05.PNG">

Reply:
Now I'm really interested how this works. Is there a convention the runtime scans for? This is totally worth documenting somewhere, if it isn't already.